modules/core/src/main/java/org/apache/ignite/internal/processors/cache/distributed/dht/preloader/GridDhtPartitionsExchangeFuture.java

```diff
@@ -769,7 +769,7 @@ public void init(boolean newCrd) throws IgniteInterruptedCheckedException {
 
             boolean crdNode = crd != null && crd.isLocal();
 
-            exchCtx = new ExchangeContext(crdNode, this);
+            exchCtx = new ExchangeContext(crdNode, this, log);
```

Comment:
Better case is to pass context or to extract it from future inplace with subsequent log field initialization.

Reply:
thanks